### PR TITLE
[Core] Add ray[adag] option to pip install

### LINKED ISSUE #46945: [core][experimental] Add ray[adag] option to pip install
### Description

Add aDAG-specific dependencies to a `pip install ray[adag]` option, so that dependencies like cupy/NCCL can be installed separately.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by ruisearch42**
## Why are these changes needed?

Add ray[adag] option to pip install

## Related issue number
Closes https://github.com/ray-project/ray/issues/46945

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/collective/requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-cupy-cuda100
\ No newline at end of file
+cupy-cuda12x
```
Comment (rkooo567) [inline python/ray/util/collective/requirements.txt:1]: Hmm it is kind of backward incompatible changes though I don't think anyone uses our collective apis at this point. should we create a separate requirement.txt for adag?

File: python/requirements/ml/dl-cpu-requirements.txt [hunk 2]
```diff
@@ -26,4 +26,4 @@ torch-cluster==1.6.3
 torch-spline-conv==1.2.2
 torch-geometric==2.5.3
 
-cupy-cuda12x==13.1.0; sys_platform != 'darwin'
```
Comment (rkooo567) [inline python/requirements/ml/dl-cpu-requirements.txt:29]: let's not change these files?

**Review by rkooo567 — COMMENTED** [review]
looks pretty good. have you manually tested it btw?

File: python/requirements/ml/dl-cpu-requirements.txt (hunk 2 above)
Comment (ruisearch42) [reply python/requirements/ml/dl-cpu-requirements.txt]: Yes, this is removed by accident

**Comment by ruisearch42** [comment]
> looks pretty good. have you manually tested it btw?

Tested with a local pip install with adag option and it works fine.

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
@angelinalg can you approve this PR? there is a doc change that needs your approval

File: python/ray/util/collective/requirements.txt (hunk 1 above)
Comment (ruisearch42) [reply python/ray/util/collective/requirements.txt]: Removed this change.

**Review by angelinalg — APPROVED** [review]
stamp

**Review by edoakes — APPROVED** [review]

**Pull request merged**